nuitka/build/static_src/CompiledCodeHelpers.c

```diff
@@ -561,7 +561,11 @@ PyObject *BUILTIN_ABS(PyObject *o) {
         return m->nb_absolute(o);
     }
 
-    return PyErr_Format(PyExc_TypeError, "bad operand type for abs(): '%.200s'", PyString_AsString(o));
+    #if PYTHON_VERSION >= 300
+        return PyErr_Format(PyExc_TypeError, "bad operand type for abs(): '%.200s'", Py_TYPE(o));
+    #else
+        return PyErr_Format(PyExc_TypeError, "bad operand type for abs(): '%.200s'", PyString_AsString(o));
```

Comment:
I don't belive that, Python never gives object `str` for exception message. Is this from the CPython original code? I would for these always check them first.

Reply:
Originally in the CPython implementation it was `return type_error("bad operand type for abs(): '%.200s'", o)`

Reply:
I wouldn't be surprised of `type_error` did do the `Py_TYPE`  on `o`, does it? Since it's a frequent pattern, they probably created a function that does it.

Reply:
Yes, You are right. And I have checked for `CPython27` and there also they are doing `Py_Type` on `o`. But when I compile with `Py27`, I get the following warning:

```
abs_test.build/CompiledCodeHelpers.c: In function ‘BUILTIN_ABS’:
abs_test.build/CompiledCodeHelpers.c:565:46: warning: format ‘%s’ expects argument of type ‘char *’, but argument 3 has type ‘struct _typeobject *’ [-Wformat=]
         return PyErr_Format(PyExc_TypeError, "bad operand type for abs():
```

Reply:
I am nearly sure t also looks up the `->tp_name` of that `PyTypeObject *`, which is indeed a string, and what is normally output in error messages.